src/network/service.rs

```diff
@@ -277,12 +284,13 @@ where
                     max_notification_size: 16 * 1024 * 1024,
                 }))
                 .chain({
-                    // The `has_grandpa_protocol` flag controls whether the chain uses GrandPa.
-                    // Note, however, that GrandPa is technically left enabled (but unused) on all
-                    // chains, in order to make the rest of the code of this module more
-                    // comprehensible.
                     iter::once(peers::NotificationProtocolConfig {
-                        protocol_name: "/paritytech/grandpa/1".to_string(),
+                        protocol_name: match &chain.fork_id {
+                            Some(id) => {
+                                format!("/{}/{}/grandpa/1", HashDisplay(&chain.genesis_hash), id)
```

Comment:
@tomaka am I correct to include `genesis_hash` here?

Reply:
Yes